Makefile

```diff
@@ -408,10 +429,10 @@ endif
 	docker push $(IMAGE_NAMESPACE)/argocli:$(IMAGE_TAG)
 	docker push $(IMAGE_NAMESPACE)/argoexec:$(IMAGE_TAG)
 	docker push $(IMAGE_NAMESPACE)/workflow-controller:$(IMAGE_TAG)
-ifeq ($(SNAPSHOT),false)
+ifeq ($(findstring release,$(GIT_BRANCH)),release)
 	# Push changes to Git
-	git push
-	git push $(VERSION)
+	git push upstream
+	git tag push $(VERSION)
```

Comment:
it should be `git push upstream  $(VERSION)` for pushing tag

Reply:
I bet we'll need to iterate this.

Reply:
git tag push $(VERSION) this will create 'push' as tag
To push a single tag:

`git push origin <tag_name>`
And the following command should push all tags (not recommended):

`git push --tags`